### PR TITLE
Remove displayed_dive dependency of ExtraDataModel

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

The main goal is to display data of current_dive instead of displayed_dive for consistency and so that we can move displayed_dive to the planner. Thus, there is no risk of forgetting to copy changes to displayed_dive.

While doing so, perform a few assorted cleanups for consistency with the other model code.

### Release note:

No - should not be user visible.

### CODE REVIEW

File: qt-models/divecomputerextradatamodel.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ void ExtraDataModel::clear()
 
 QVariant ExtraDataModel::data(const QModelIndex &index, int role) const
 {
-	struct extra_data *ed = get_dive_dc(&displayed_dive, dc_number)->extra_data;
+	struct extra_data *ed = dc ? dc->extra_data : nullptr;
```
Comment (dirkhh) [inline qt-models/divecomputerextradatamodel.cpp]: what happens to us if the `dc` pointer becomes invalid? Do we know that any operation that could reallocate our dive table clears the model? Specifically, I don't see any code that resets the `dc` variable to nullptr.

**Review by dirkhh — COMMENTED** [review]
Maybe I don't understand the lifecycle of this `ExtraDataModel`, but it looks to me like we are creating dangling pointers here...

**Review by dirkhh — APPROVED** [review]
so assuming we do the right thing and call `clear()` when things change (which is what you have been diligently working on), this looks good now.

**Comment by bstoeger** [comment]
In principle yes - however, I will change this to cache the QStrings. Thus, no potential of dangling pointers.

**Comment by dirkhh** [comment]
> In principle yes - however, I will change this to cache the QStrings. Thus, no potential of dangling pointers.

cool, so I'll hold off on merging.

**Comment by bstoeger** [comment]
Data now cached.

**Review by dirkhh — APPROVED** [review]
I really like this approach.

**Pull request merged**